lib/core/src/getRodsEnv.cpp

```diff
@@ -139,12 +142,17 @@ extern "C" {
         createRodsEnvDefaults( &rodsEnvArg );
     }
 
-    static
-    int capture_string_property(
-        const std::string&             _key,
-        char*                          _val ) {
+    static int capture_string_property(const std::string& _key, char* _val, size_t _val_size)
+    {
         try {
-            sprintf(_val, "%s", irods::get_environment_property<const std::string>(_key).c_str());
+            auto property = irods::get_environment_property<const std::string>(_key);
+
+            // If _val is undefined, _val_size is zero, or truncation would
+            // occur because property.size() >= _val_size, return an error
+            if (!_val || !_val_size || property.size() >= _val_size) {
```

Comment:
Consider doing the pointer and size checks for `_val` and `_val_size` before L148.

The check comparing the size of the property and `_val_size` can stay here.